rust/connlib/tunnel/src/lib.rs

```diff
@@ -116,7 +116,9 @@ where
             }
 
             if let Some(dns_query) = self.role_state.poll_dns_queries() {
-                self.io.perform_dns_query(dns_query);
+                if let Err(e) = self.io.perform_dns_query(dns_query.clone()) {
+                    self.role_state.on_dns_result(dns_query, Err(e))
```

Comment:
This is where those two functions with similar signature changes get hooked together.